### PR TITLE
NorSand use pdstrain and make some parameters optional

### PR DESCRIPTION
**Pull request by ezrayst**
**Describe the PR**
NorSand model has been modified:
- To use `pdstrain` within `mpm::materials`
- As discussed, `bonded_model` is now an optional input variable along with the other five variables including `p_cohesion_initial`, `p_dilation_initial`, `m_cohesion`, `m_dilation` and `m_modulus`.
- The test cases have also been changed to reflect this change, only for the non bonded part of the test

### CODE REVIEW

File: include/materials/norsand.tcc [hunk 1]
```diff
@@ -35,19 +35,32 @@ mpm::NorSand<Tdim>::NorSand(unsigned id, const Json& material_properties)
...
-        material_properties.at("p_dilation_initial").template get<double>();
-    // Cohesion degradation parameter m upon shearing
-    m_cohesion_ = material_properties.at("m_cohesion").template get<double>();
-    // Dilation degradation parameter m upon shearing
-    m_dilation_ = material_properties.at("m_dilation").template get<double>();
-    // Parameter for shear modulus
-    m_modulus_ = material_properties.at("m_modulus").template get<double>();
+    if (material_properties.find("bond_model") != material_properties.end()) {
+      bond_model_ = material_properties.at("bond_model").template get<bool>();
+      // Initial p_cohesion
```
Comment (kks32) [inline include/materials/norsand.tcc:43]: The following should only be checked if `bond_model_` is `true`

File: tests/materials/norsand_test.cc [hunk 2]
```diff
@@ -39,13 +39,6 @@ TEST_CASE("NorSand is checked in 3D non-bonded model",
   jmaterial["hardening_modulus"] = 200.0;
   jmaterial["void_ratio_initial"] = 0.85;
   jmaterial["p_image_initial"] = 87014.6;
-  jmaterial["bond_model"] = false;
```
Comment (kks32) [inline tests/materials/norsand_test.cc:42]: Why are they removed from testing?

**Review by kks32 — CHANGES_REQUESTED** [review]

File: tests/materials/norsand_test.cc (hunk 2 above)
Comment (ezrayst) [reply tests/materials/norsand_test.cc:42]: To test that they can be initialized without specifying those `bonded` parameters

File: include/materials/norsand.tcc (hunk 1 above)
Comment (ezrayst) [reply include/materials/norsand.tcc:43]: Modified!

File: include/materials/norsand.tcc [hunk 3]
```diff
@@ -35,19 +35,35 @@ mpm::NorSand<Tdim>::NorSand(unsigned id, const Json& material_properties)
...
+      p_dilation_initial_ =
+          material_properties.at("p_dilation_initial").template get<double>();
+      // Cohesion degradation parameter m upon shearing
+      m_cohesion_ = material_properties.at("m_cohesion").template get<double>();
+      // Dilation degradation parameter m upon shearing
+      m_dilation_ = material_properties.at("m_dilation").template get<double>();
+      // Parameter for shear modulus
+      m_modulus_ = material_properties.at("m_modulus").template get<double>();
+    } else {
+      // Initial p_cohesion
```
Comment (kks32) [inline include/materials/norsand.tcc]: Could we set these as the default values in the header and we don't need an else clause

File: tests/materials/norsand_test.cc (hunk 2 above)
Comment (kks32) [reply tests/materials/norsand_test.cc:42]: Are there tests with these values?

**Review by kks32 — CHANGES_REQUESTED** [review]

File: tests/materials/norsand_test.cc (hunk 2 above)
Comment (ezrayst) [reply tests/materials/norsand_test.cc:42]: Test for `false`? The default initialization is `false` anyway right?

**Review by bodhinandach — APPROVED** [review]

**Review by kks32 — APPROVED** [review]

**Pull request merged**